magic_bridge/ic/src/dip20_proxy/src/proxy.rs

```diff
@@ -412,6 +413,19 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_principal_from_nat() {
+        let eth_address_as_nat =
+            Nat::from_str("1118288024408649503359660893691376548931478070077").unwrap();
```

Comment:
This test covers one type of address. Addresses starting with a `0` would have problems.

```
    #[test]
    fn principal_padding() {
        let slice =
            hex::decode("B2BF35A84FAC4062A1C0BC4F8891A4AF09C5E05E4155CAE6355B2402").unwrap();

        let n = Nat::from(num_bigint::BigUint::from_bytes_be(&slice[..]));

        assert_eq!(slice.len(), 28);

        let mut bytes: [u8; 29] = [0; 29];
        fill_from_slice(&mut bytes, &n.0.to_bytes_le().as_slice());
        bytes.reverse();
        let new_slice = bytes.to_vec();

        let p = Principal::from_slice(&new_slice);

        let p_expected = "kyxzn-5aawk-7tlkc-pvrag-fioax-rhyre-nev4e-4lyc6-ifk4v-zrvlm-sae";

        assert_eq!(p.to_text(), p_expected);
    }
```
you must verify that a test like this works.

More notes:

The issues is endianness between ethereum and IC tooling. We need to pad manually the length of the principal which is `29` bytes for a normal principal and `10` bytes for a canister principal